staging/src/k8s.io/client-go/dynamic/dynamicinformer/interface.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamicinformer
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/informers"
+)
+
+type DynamicSharedInformerFactory interface {
+	Start(stopCh <-chan struct{})
+	ForResource(gvr schema.GroupVersionResource) informers.GenericInformer
+	WaitForCacheSync(stopCh <-chan struct{}) map[string]bool
+}
+
+type TweakListOptionsFunc func(*metav1.ListOptions)
```

Comment:
are you okay with defining this func here ? 
I think it is already defined somewhere here `k8s.io/client-go/informers/internalinterfaces`

Reply:
> are you okay with defining this func here ?
> I think it is already defined somewhere here `k8s.io/client-go/informers/internalinterfaces`

Yes, we want to keep this here.  Eventually we are likely to need to snip the links to the typed clients.